mvc/controllers/actions.rst

```diff
@@ -1,8 +1,63 @@
+ 
+Defining Actions
+----------------
+Any public method on a controller type is an action. Parameters on actions are bound to request data and validated using :doc:`model binding <../models/model-binding>`. Any action that accepts parameters should verify that the incoming data is valid by checking ``ModelState.IsValid``.
+
+.. warning:: Action methods that accept parameters should always check ``ModelState.IsValid`` as part of their processing.
+
+Action methods should contain logic for mapping an incoming request to a business concern. These business concerns should typically be represented as services that your controller accesses through dependency injection. Actions then map the result of the business action to an application state.
+
+Actions can return an ``IActionResult`` that produces a response. You can either create and return your own ``ActionResult`` directly, or you can use one of the many helper methods available on the Controller base class.
+
+Controller Helper Methods
+#########################
+
+Inheriting from the base ``Controller`` class provides the following helper methods for returning various kinds of responses.
+
+View
+	Returns a View that uses a model to render HTML. Example: ``return View(customer);``
+
+HTTP Status Code
+	Return an HTTP status code. Examples: ``return Ok();`` or ``return BadRequest();``
+
+Formatted Response
+	Return ``Json`` or similar to format an object in a specific manner. Example: ``return Json(customer);``
+
+An object
+	Return an object you want formatted based on the client's request. Learn more about :doc:`../models/formatting`
+
+Cross-Cutting Concerns
+######################
+
+In most apps, many actions will share parts of their workflow. For instance, most of an app might be available only to authenticated users, or might benefit from caching. Try to keep your actions from growing too large by using :doc:`filters` to handle these concerns. This can help eliminate duplication within your actions, allowing them to follow the `Don't Repeat Yourself (DRY) principle <http://deviq.com/don-t-repeat-yourself/>`_.
+
+In the case of authorization and authentication, you can apply the ``Authorize`` attribute to any actions that require it. Adding this attribute will ensure the appropriate ``ActionFilter`` is applied to any request for this action. Attributes can also be applied at the ``Controller`` level, so groups of actions with similar behavior can use controller-level attributes to apply the same filters to every action. Action-level attributes will override controller-level attributes, so for example a controller could be marked with the ``Authorize`` attribute, but one of its actions could override this with the ``AllowAnonymous`` attribute. Learn more about :doc:`../security/authorization-filters`.
```

Comment:
This should link to the doc on filters rather than going too in depth here. For instance 

> Action-level attributes will override controller-level attributes

This isn't true in the general case. This is true for `[Authorize]`